test-harness/src/matcher/base_matcher.rs

```diff
@@ -214,13 +225,17 @@ impl Matcher for BaseMatcher {
             }
         }
 
-        if let Some(leaf_span_matcher) = &self.leaf_span {
-            match event.spans().last() {
-                Some(leaf_span) if leaf_span_matcher.matches(leaf_span) => {}
-                _ => {
-                    // Expected a leaf span, but the event is missing a leaf span or doesn't have
-                    // the correct leaf span.
-                    return Ok(false);
+        if !self.leaf_spans.is_empty() {
```

Comment:
Is this check necessary?  I think the code behaves the same without it (since the `for` loop will do nothing if `self.leaf_spans` is empty

Reply:
Good point. Removed this check (and the one for the loop above it).